internal/gengapic/gengapic_test.go

```diff
@@ -499,3 +499,46 @@ func Test_isLRO(t *testing.T) {
 		}
 	}
 }
+
+func Test_transportParse(t *testing.T) {
+	for _, tst := range []struct {
+		param        string
+		expectedOpts options
+		expectErr    bool
+	}{
+		{
+			param: "transport=grpc,go-gapic-package=path;pkg",
+			expectedOpts: options{
+				transports: []Transport{grpc},
+				pkgPath:    "path",
+				pkgName:    "pkg",
+				outDir:     "path",
+			},
+			expectErr: false,
+		},
+		{
+			param: "transport=rest+grpc,go-gapic-package=path;pkg",
+			expectedOpts: options{
+				transports: []Transport{rest, grpc},
+				pkgPath:    "path",
+				pkgName:    "pkg",
+				outDir:     "path",
+			},
+			expectErr: false,
+		},
+		{
+			param:        "transport=tcp,go-gapic-package=path;pkg",
+			expectedOpts: options{},
+			expectErr:    true,
+		},
+	} {
+		opts, err := ParseOptions(&tst.param)
+		if tst.expectErr {
+			if err == nil {
```

Comment:
`if tst.expectErr && (err == nil)` doesn't work? Weird....

Reply:
Suggestion for debugging this: have the test print `("%v", err)`. I worry this may hide a simple bug, so we should try to resolve this.

Reply:
I think I figured it out: the test struct is storing the expected options by value, but ParseOptions is returning by pointer, so the comparison fails because sometimes the returned options is nil, and the expected options cannot be nil. Changing the expected options field to a pointer fixed this.